nipoppy_cli/nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +121,99 @@ def run_main(self):
+                message="Nipoppy layout initialized.",
+            )
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
 
+    def _init_as_datalad_dataset(self) -> None:
+        if self.use_dalatad:
+            from datalad import api
+
+            api.create(path=self.dpath_root, result_renderer="disabled")
+            self._make_git_attributes()
+            self._make_git_ignore()
+
+    def _init_manifest_from_bids_dataset(self) -> None:
+        """Assume a BIDS dataset with session level folders.
+
+        No BIDS validation is done.
+        """
+        df = {"participant_id": [], "visit_id": [], "session_id": [], "datatype": []}
+        participant_ids = sorted(
+            [
+                x.name
+                for x in (self.dpath_root / "bids").iterdir()
+                if x.is_dir() and x.name.startswith("sub-")
+            ]
+        )
+
+        self.logger.info("Creating a manifest.csv from the BIDS dataset content.")
+
+        for ppt in participant_ids:
+
+            session_ids = sorted(
+                [
+                    x.name
+                    for x in (self.dpath_root / "bids" / ppt).iterdir()
+                    if x.is_dir() and x.name.startswith("ses-")
+                ]
+            )
+            if not session_ids:
+                self.logger.warning(
+                    f"Skipping subject '{ppt}': could not find a session level folder."
+                )
+                continue
+
+            for ses in session_ids:
+                datatypes = sorted(
+                    [
+                        x.name
+                        for x in (self.dpath_root / "bids" / ppt / ses).iterdir()
+                        if x.is_dir()
+                    ]
+                )
+
+                df["participant_id"].append(ppt.replace("sub-", ""))
+                df["session_id"].append(ses.replace("ses-", ""))
+                df["datatype"].append("[" + "'" + "' ,'".join(datatypes) + "'" + "]")
+
+        df["visit_id"] = df["session_id"]
+
+        df = pd.DataFrame(df)
+        df.to_csv(self.dpath_root / "manifest.csv", index=False)
+
+    def _make_git_attributes(self) -> None:
+        CONTENT = [
+            "* annex.backend=MD5E",
+            "**/.git* annex.largefiles=nothing",
+            "*.csv annex.largefiles=nothing",
+            "*.tsv annex.largefiles=nothing",
+            "*.json annex.largefiles=nothing",
+            "**/README.md annex.largefiles=nothing",
+        ]
+        with open(self.dpath_root / ".gitattributes", "w") as f:
+            for line in CONTENT:
+                f.write(f"{line}\n")
+
+    def _make_git_ignore(self) -> None:
+        CONTENT = ["sourcedata", "scratch", "proc/pybids/bids_db"]
```

Comment:
wonder if more things should not be ignore by default

for example I think that each derivative dataset should be ignore or a subdataset otherwise you end up with datalad datasets that contain too many things